internal/server/handler.go

```diff
@@ -96,10 +96,14 @@ func (s *Server) handler() http.HandlerFunc {
 		}
 
 		// Need to sign the header, just before sending it
-		r = signer.SignV4(*r, s.cfg.S3Settings.AccessKeyID,
-			s.cfg.S3Settings.SecretAccessKey,
-			val.SessionToken,
-			s.cfg.S3Settings.Region)
+		if s.cfg.S3Settings.AccessKeyID == "" && s.cfg.S3Settings.SecretAccessKey == "" {
```

Comment:
nit: should this be `or` instead of `and`? I imagine that if either aren't set, we shouldn't be using them?

Reply:
Yes, I agreed in the first version I put `or` but I saw a similar code in Mattermost using `and`  https://github.com/mattermost/mattermost/blob/c113cdbd9bc2629081b309d2cf6edc50368a4732/server/platform/shared/filestore/s3store.go#L150 . But I think `or`makes more sense. Thanks

Reply:
@nickmisasi Thanks for your review I added some changes